gitlab/data_source_gitlab_branch.go

```diff
@@ -0,0 +1,133 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/xanzy/go-gitlab"
+)
+
+func dataSourceGitlabBranch() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGitlabBranchRead,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"web_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"default": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"can_push": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"merged": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"commit": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
```

Comment:
What is the purpose of the optional attributes under here, if the schema is computed?